apps/hash-ai-worker-ts/src/activities/embeddings.ts

```diff
@@ -0,0 +1,75 @@
+import type {
+  BaseUrl,
+  EntityPropertiesObject,
+  EntityPropertyValue,
+  PropertyTypeWithMetadata,
+} from "@local/hash-subgraph";
+import OpenAI from "openai/index";
+import { CreateEmbeddingResponse } from "openai/resources";
+import Usage = CreateEmbeddingResponse.Usage;
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+const createEmbeddingInput = (params: {
+  propertyType: PropertyTypeWithMetadata;
+  property: EntityPropertyValue;
+}): string => {
+  return `${params.propertyType.schema.title}: ${JSON.stringify(
+    params.property,
+  )}`;
+};
+
+export const createEmbeddings = async (params: {
+  entityProperties: EntityPropertiesObject;
+  propertyTypes: PropertyTypeWithMetadata[];
+}): Promise<{
+  embeddings: { property?: BaseUrl; embedding: number[] }[];
+  usage: Usage;
+}> => {
+  if (params.propertyTypes.length === 0) {
+    return {
+      embeddings: [],
+      usage: {
+        prompt_tokens: 0,
+        total_tokens: 0,
+      },
+    };
+  }
+
+  // sort property types by their base url
+  params.propertyTypes.sort((a, b) =>
+    a.metadata.recordId.baseUrl.localeCompare(b.metadata.recordId.baseUrl),
+  );
+
+  const input = [""];
```

Comment:
I think we could make it easier to understand what's going on here without needing to know how it works already. Either via splitting out the variables, e.g.
```ts
  const propertyEmbeddings = [];
  let combinedEntityEmbedding = "";
  for (const propertyType of params.propertyTypes) {
    ...
    combinedEntityEmbedding += `${embeddingInput}\n`;
    propertyEmbeddings.push(embeddingInput);
  }

  const response = await openai.embeddings.create({
    input: [combinedEntityEmbedding, ...propertyEmbeddings],
```

or adding a comment, something like
```suggestion
  /**
   * We want to create embeddings for:
   * 1. Each individual '[Property Title]: [Value]' pair, and
   * 2. A list of all property key:value pairs
   * 
   * We use the first item in the array to store the combined 'all properties' list.
   */
  const input = [""];
```